### ISSUE TITLE
BYD firmware version wrong on vrm.

### ISSUE THREAD

**Issue by mpvader** [victronenergy/venus#229]
The same canbus data (`can0  35F   [8]  4C 69 04 0F 32 00 11 17`) shows as hex on vrm: v4.0F. While in the Ccgx it is shown in decimals.

Most probably this needs fixing in the Venus OS, fix is expected in the next v2.12~ test version.
![96633b9f-5393-4e85-a668-b4180ee26cfd](https://user-images.githubusercontent.com/5200296/33421472-76e65f18-d5b2-11e7-9d6e-e9325d5b9939.png)
![637da620-de7b-4550-a360-0b33866c54dc](https://user-images.githubusercontent.com/5200296/33421473-77178c8c-d5b2-11e7-9bb0-b01e1a5446f0.jpeg)

**Comment by mpvader** [comment]
The gui shows it correctly because it does a gettext.

BYD battery:
```
dbus -y com.victronenergy.battery.socketcan_can0 /FirmwareVersion GetValue
value = 1039   (= 0x40F => v4.0F in Victron notation, but BYD expects v4.15)
dbus -y com.victronenergy.battery.socketcan_can0 /FirmwareVersion GetText 
value = 'v4.15'
```

BMV:
```
# dbus -y com.victronenergy.battery.ttyO2 /FirmwareVersion GetValue
value = 776  (= 0x308 => v3.08 in Victron notation)
root@ccgx:~# dbus -y com.victronenergy.battery.ttyO2 /FirmwareVersion GetText 
value = 'v3.08'
```

One solution is just to no longer do the conversion from number to string in vrmlogger, but use the GetText for this path instead. (in the vrm database we store firmware values as formatted string).

@jhofstee any thoughts?

**Comment by mpvader** [comment]
Conversion by vrmlogger is done here:

https://git.victronenergy.com/ccgx/dbus_vrm/blob/master/conversions.py#L23

and here:

https://github.com/victronenergy/velib_python/blob/master/ve_utils.py#L55

**Comment by mpvader** [comment]
As discussed with @jhofstee, indeed the proper fix is to use the GetText value, and not the GetValue.

Which maybe means adding GetText values to dbus-monitor, and renaming functions like [these](https://github.com/victronenergy/velib_python/blob/master/dbusmonitor.py#L200) to handle_properties_changed().

**Comment by izak** [comment]
Solution we decided on is to use a kind of HED (hex-coded decimal, coined from the similar term BCD).

Afterwards: 
```
root@ccgx:/opt/victronenergy/can-bus-bms# dbus -y com.victronenergy.battery.socketcan_can0 /FirmwareVersion GetText 
value = 'v4.15'
root@ccgx:/opt/victronenergy/can-bus-bms# dbus -y com.victronenergy.battery.socketcan_can0 /FirmwareVersion GetValue
value = 1045
```

1045 translates to 4.21 in decimal, which translates to 4.15 when rendered in hex.

**Comment by mpvader** [comment]
Fixed in vrmlogger 2.90 & 2.91. waiting for inclusion in Venus

**Comment by mpvader** [comment]
This was in v2.12 for a while, but then reverted because of issues. Is now scheduled for v2.13
